dynamodb/src/main/scala/monix/connect/dynamodb/DynamoDbSubscriber.scala

```diff
@@ -17,45 +17,77 @@
 
 package monix.connect.dynamodb
 
-import monix.eval.Task
 import monix.execution.cancelables.AssignableCancelable
 import monix.execution.{Ack, Callback, Scheduler}
-import monix.reactive.observers.Subscriber
 import monix.reactive.Consumer
 import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
 import software.amazon.awssdk.services.dynamodb.model.{DynamoDbRequest, DynamoDbResponse}
+import monix.reactive.observers.Subscriber
 
 import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
 
-private[dynamodb] class DynamoDbSubscriber[In <: DynamoDbRequest, Out <: DynamoDbResponse]()(
+/**
+  * A pre-built [[Consumer]] implementation that expects incoming [[DynamoDbRequest]] and
+  * executes them with a retryable-with-delay strategy that makes it more flexible to unexpected failures.
+  *
+  * @param retriesPerOp indicates the number of times that an operation would be retried in case there was a failure
+  *                     it must be higher or equal than 0.
+  * @param dynamoDbOp an implicit [[DynamoDbOp]] that would abstract the execution of the specific operation.
+  * @param delayAfterFailure delay after failure for the execution of a single [[DynamoDbOp]].
+  * @param client an implicit instance of a [[DynamoDbAsyncClient]].
+  * @tparam In a lower bounded type of [[DynamoDbRequest]] that represents the incoming elements.
+  * @tparam Out The response of the execution as type parameter lower bounded by [[DynamoDbResponse]].
+  */
+private[dynamodb] class DynamoDbSubscriber[In <: DynamoDbRequest, Out <: DynamoDbResponse](
+  retriesPerOp: Int,
+  delayAfterFailure: Option[FiniteDuration])(
   implicit
   dynamoDbOp: DynamoDbOp[In, Out],
   client: DynamoDbAsyncClient)
   extends Consumer[In, Out] {
 
+  require(retriesPerOp >= 0, "Retries per operation must be higher or equal than 0.")
+
   override def createSubscriber(cb: Callback[Throwable, Out], s: Scheduler): (Subscriber[In], AssignableCancelable) = {
     val sub = new Subscriber[In] {
 
       implicit val scheduler = s
-      private var dynamoDbResponse: Task[Out] = _
-
-      def onNext(dynamoDbRequest: In): Future[Ack] = {
-        dynamoDbResponse = Task.from(dynamoDbOp.execute(dynamoDbRequest))
+      private var dynamoDbResponse: Out = _
```

Comment:
btw what happens if `Observable` ends without emitting any requests?

Reply:
good question, it does return `null` :(
we could fix that by returning an `Option`, but maybe just `Unit`, what do you think?

Reply:
now it materialises to `Unit`.